### DISCUSSION TITLE
Differentiate between dependencies and devDependencies in `dependsOn`?

### DISCUSSION THREAD

**Discussion by GeorgeTaveras1231** [vercel/turborepo#2005] [Help] [answered]
For reference, this is my config:

```json5
{
  "$schema": "https://turborepo.org/schema.json",
  "pipeline": {
    "build.pkg": {
      // A package's `build` script depends on that package's
      // dependencies and devDependencies
      // `build` tasks  being completed first
      // (the `^` symbol signifies `upstream`).
      "dependsOn": ["^build.pkg"],
      // note: output globs are relative to each package's `package.json`
      // (and not the monorepo root)
      "outputs": ["lib/**"],
      "inputs": ["src/**"]
    }
  }
}
```

Much of my repository are packages. When they have a dependency, most of the time, that dependency is not bundled into the package -- so changes to the `dependencies` config (in package json) which we use to denote **runtime** dependencies, don't change the output of the `build.pkg` command. However, `devDependencies` include testing dependencies but also **build** dependencies. Some of those **build** dependencies are internally developed, and when changed, can actually cause the output of `build.pkg` to change.

So my problem is this. I need a way to differentiate between **build** dependencies, and **runtime** dependencies. 

I've considered creating a separate script for internal packages that are also (generally) build tools, however, this feels like a hack, since a given package can be a **build** dependency or a **runtime** dependency, ultimately it depends on the consumer of the package to decide how to use it (wether the code in the dependency is bundled, referenced during the build process, or just referenced during runtime).

Is there currently a good way to create the distinction between **build** dependencies and **runtime** dependencies? maybe by allowing making a `devDependencies` only upstream reference in `dependsOn`?

**Comment by nathanhammond** [comment, marked as answer]
See my answer here: https://github.com/vercel/turborepo/discussions/2006#discussioncomment-3686001
